gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -144,6 +144,8 @@
 
   private static final String LIST_OBJECT_FIELDS_FORMAT = "items(%s),prefixes,nextPageToken";
 
+  private MetricsPublisher metricsPublisher = new NoOpMetricsPublisher();
```

Comment:
Probably better to make it final, and assign no-op implementation if metrics are disabled.